src/Azure/Orleans.Streaming.EventHubs/Providers/EventDataGeneratorStreamProvider/EventHubPartitionDataGenerator.cs

```diff
@@ -45,26 +46,30 @@ public bool TryReadEvents(int maxCount, out IEnumerable<EventData> events)
                 return false;
             }
             int count = maxCount;
-            List<EventData> eventDataList = new List<EventData>();
+            List<EventData> eventDataList = new List<EventData>(maxCount);
             while (count-- > 0)
             {
                 this.SequenceNumberCounter.Increment();
 
-                var eventData = EventHubBatchContainer.ToEventData(
+                // Create an EventData instance with an empty body. The body will be set later
+                // from the batch container's context. Because there is a need to explicitly set
+                // broker-owned properties such as the offset, sequence number, and partition key,
+                // an instance is created using the model factory, which avoids the need to set
+                // directly via the underlying AMQP message.
+                var eventData = EventHubsModelFactory.EventData(
+                        eventBody: BinaryData.Empty,
+                        partitionKey: StreamId.GetKeyAsString(),
+                        offsetString: DateTime.UtcNow.Ticks.ToString(CultureInfo.InvariantCulture),
+                        sequenceNumber: this.SequenceNumberCounter.Value);
+
+                EventHubBatchContainer.UpdateEventData(
+                    eventData,
                     this.serializer,
                     this.StreamId,
                     GenerateEvent(this.SequenceNumberCounter.Value),
                     RequestContextExtensions.Export(this.deepCopier));
 
-               var wrapper = new WrappedEventData(
-                    eventData.Body,
-                    eventData.Properties,
-                    eventData.SystemProperties,
```

Comment:
You don't want to override the system properties dictionary, as that's a path for mocked test data and bypasses the underlying AMQP message.   Likely not a concern in this context, as you can't round-trip that data anyway, but there's really no value in setting it.